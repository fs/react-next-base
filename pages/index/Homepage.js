import React from 'react';
import styled from 'styled-components';
import DefaultTemplate from 'components/templates/DefaultTemplate';

const Title = styled.h1`
  font-size: 2rem;
  text-align: center;
`;

const Home = () => (
  <DefaultTemplate>
    <Title>Welcome to React Next Base</Title>
  </DefaultTemplate>
);

export default Home;
