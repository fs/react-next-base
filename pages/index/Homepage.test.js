import React from 'react';
import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';
import renderWithTheme from '__tests__/helpers/renderWithTheme';

import Homepage from '.';

describe('Homepage', () => {
  afterEach(() => {
    jest.clearAllMocks();
    jest.resetModules();
    cleanup();
  });

  test('should render correctly', () => {
    // Arrange

    // Act
    const { container } = render(renderWithTheme(<Homepage />));

    // Assert
    expect(container).toMatchSnapshot();
  });
});
