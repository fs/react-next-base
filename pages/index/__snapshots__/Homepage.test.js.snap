// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
}

.c1 {
  font-size: 2rem;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      Welcome to React Next Base
    </h1>
  </div>
</div>
`;
